packages/umi-serializer-beet/src/createBeetSerializer.ts

```diff
@@ -88,14 +112,24 @@ export function createBeetSerializer(
       },
       deserialize: (bytes: Uint8Array, offset = 0) => {
         if (typeof size === 'object' && bytes.slice(offset).length === 0) {
-          return handleEmptyBuffer<U[]>('array', [], offset);
+          throw new DeserializingEmptyBufferError('array');
         }
         const [resolvedSize, newOffset] = getResolvedSize(
           size,
           [item.fixedSize],
           bytes,
           offset
         );
+        if (
+          typeof size === 'number' &&
+          bytes.slice(offset).length < resolvedSize
+        ) {
+          throw new NotEnoughBytesError(
+            'array',
+            resolvedSize,
+            bytes.slice(offset).length
+          );
```

Comment:
Added this code to implement what we talked about earlier (fixed size directives needing to override ‘tolerance’)